File: app/src/main/java/com/example/food_app/activity/CartActivity.java
package com.example.food_app.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.food_app.adapter.CartAdapter;
import com.example.food_app.databinding.ActivityCartBinding;
import com.example.food_app.helper.ChangeNumberItemsListener;
import com.example.food_app.helper.ManagmentCart;

public class CartActivity extends AppCompatActivity {
    ActivityCartBinding binding;
    private ManagmentCart managmentCart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityCartBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        managmentCart = new ManagmentCart(this);
        setVariable();
        calculateCart();
        initCartList();
    }

    private void initCartList() {
        if (managmentCart.getListCart().isEmpty()) {
            binding.emptyTxt.setVisibility(View.VISIBLE);
            binding.scrollViewCart.setVisibility(View.GONE);
        } else {
            binding.emptyTxt.setVisibility(View.GONE);
            binding.scrollViewCart.setVisibility(View.VISIBLE);
        }
        binding.cartView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        binding.cartView.setAdapter(new CartAdapter(managmentCart.getListCart(), managmentCart, new ChangeNumberItemsListener() {
            @Override
            public void change() {
                calculateCart();
            }
        }));
    }

    private void calculateCart() {
        double percentTax = 0.02;
        double delivery = 10;
        double tax = Math.round(managmentCart.getTotalFee() * percentTax * 100.0) / 100.0;
        double total = Math.round((managmentCart.getTotalFee() + tax + delivery) * 100.0) / 100.0;
        double itemTotal = Math.round(managmentCart.getTotalFee() * 100.0) / 100.0;

        binding.totalFeeTxt.setText("$" + itemTotal);
        binding.taxTxt.setText("$" + tax);
        binding.deliveryTxt.setText("$" + delivery);
        binding.totalTxt.setText("$" + total);
    }

    private void setVariable() {
        binding.backButton.setOnClickListener(v -> {
            startActivity(new Intent(CartActivity.this, MainActivity.class));
        });
    }
}

File: app/src/main/java/com/example/food_app/activity/CartActivity.kt
package com.example.food_app.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.food_app.adapter.CartAdapter
import com.example.food_app.databinding.ActivityCartBinding
import com.example.food_app.helper.ManagmentCart

class CartActivity : AppCompatActivity() {
    var binding: ActivityCartBinding? = null
    private var managmentCart: ManagmentCart? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityCartBinding.inflate(layoutInflater)
        setContentView(binding!!.root)

        managmentCart = ManagmentCart(this)
        setVariable()
        calculateCart()
        initCartList()
    }

    private fun initCartList() {
        if (managmentCart!!.listCart.isEmpty()) {
            binding!!.emptyTxt.visibility = View.VISIBLE
            binding!!.scrollViewCart.visibility = View.GONE
        } else {
            binding!!.emptyTxt.visibility = View.GONE
            binding!!.scrollViewCart.visibility = View.VISIBLE
        }
        binding!!.cartView.layoutManager =
            LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        binding!!.cartView.adapter = CartAdapter(
            managmentCart!!.listCart, managmentCart!!
        ) { calculateCart() }
    }

    private fun calculateCart() {
        val percentTax = 0.02
        val delivery = 10.0
        val tax = Math.round(managmentCart!!.totalFee * percentTax * 100.0) / 100.0
        val total = Math.round((managmentCart!!.totalFee + tax + delivery) * 100.0) / 100.0
        val itemTotal = Math.round(managmentCart!!.totalFee * 100.0) / 100.0

        binding!!.totalFeeTxt.text = "$$itemTotal"
        binding!!.taxTxt.text = "$$tax"
        binding!!.deliveryTxt.text = "$$delivery"
        binding!!.totalTxt.text = "$$total"
    }

    private fun setVariable() {
        binding!!.backButton.setOnClickListener { v ->
            startActivity(
                Intent(
                    this@CartActivity,
                    MainActivity::class.java
                )
            )
        }
    }
}